data-service/src/main/java/com/amazonaws/blox/dataservice/repository/model/EnvironmentDDBRecord.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.dataservice.repository.model;
+
+import com.amazonaws.blox.dataservice.model.EnvironmentHealth;
+import com.amazonaws.blox.dataservice.model.EnvironmentStatus;
+import com.amazonaws.blox.dataservice.model.EnvironmentType;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBHashKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBIndexHashKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBIndexRangeKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBRangeKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTable;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTypeConverted;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTypeConvertedEnum;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBVersionAttribute;
+import java.time.Instant;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@DynamoDBTable(tableName = "Environments")
+@Data
+@Builder
+//Required for builder because an empty constructor exists
+@AllArgsConstructor
+//Required for dynamodbmapper
+@NoArgsConstructor
+public class EnvironmentDDBRecord {
+
+  public static final String ENVIRONMENT_ID_HASH_KEY = "environmentId";
+  public static final String ENVIRONMENT_NAME_RANGE_KEY = "environmentName";
+  public static final String ENVIRONMENT_CLUSTER_GSI_NAME = "environmentClusterIndex";
+  public static final String ENVIRONMENT_CLUSTER_INDEX_HASH_KEY = "accountId";
+  public static final String ENVIRONMENT_CLUSTER_INDEX_RANGE_KEY = "cluster";
+
+  public static EnvironmentDDBRecord withKeys(
+      final String environmentId, final String environmentName) {
+
+    return EnvironmentDDBRecord.builder()
+        .environmentId(environmentId)
+        .environmentName(environmentName)
+        .build();
+  }
+
+  public static EnvironmentDDBRecord withGSIKeys(final String accountId, final String cluster) {
+    return EnvironmentDDBRecord.builder().accountId(accountId).clusterName(cluster).build();
+  }
+
+  @DynamoDBHashKey(attributeName = ENVIRONMENT_ID_HASH_KEY)
```

Comment:
Won't DDBMapper infer the column name from the field name here? Why repeat the name in a string constant?

Reply:
Yes but the field names can be different from the table values like cluster/clusterName so in general I've used attritbutenames everywhere and not just where they differ to be consistent.

Reply:
Wait, why would we want those names to be inconsistent in the first place? I mean, these `*DDBRecord` classes are *already* the at the level of abstraction where they should exactly correspond with the DB Schema. IIRC the main reason to use the explicit attribute names was to interact with legacy DBs that had e.g. `snake_case` column names.